File: src/main/java/chloe/movietalk/domain/ReviewLike.java
package chloe.movietalk.domain;

import jakarta.persistence.*;
import lombok.*;

@ToString
@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class ReviewLike extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, updatable = false)
    private SiteUser user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "review_id", nullable = false, updatable = false)
    private Review review;

    @Builder
    public ReviewLike(SiteUser user, Review review) {
        this.user = user;
        this.review = review;
    }
}

File: src/main/java/chloe/movietalk/domain/ReviewLike.kt
package chloe.movietalk.domain

import jakarta.persistence.*
import lombok.*

@ToString
@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
class ReviewLike @Builder constructor(
    @field:JoinColumn(
        name = "user_id",
        nullable = false,
        updatable = false
    ) @field:ManyToOne(fetch = FetchType.LAZY) private var user: SiteUser?, @field:JoinColumn(
        name = "review_id",
        nullable = false,
        updatable = false
    ) @field:ManyToOne(fetch = FetchType.LAZY) private var review: Review?
) : BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null
}
